### PR TITLE
Tweak: Allowing to import and export plugins as part of the kit content [ED-4978]

### PR DESCRIPTION
**Pull request by rotemee**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/app/assets/js/molecules/data-table.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+export default function DataTable( {
+	className,
+	onSelect,
+	initialSelected,
+	initialDisabled,
+	headers,
+	layout,
+	rows,
+	selection,
+	withHeader,
```
Comment (rotemee) [inline core/app/assets/js/molecules/data-table.js:14]: Check if withHeader can be removed an rely only on the headers prop.

File: core/app/assets/js/ui/atoms/checkbox.js [hunk 2]
```diff
@@ -4,15 +4,29 @@ import './checkbox.scss';
...
+	if ( props.indeterminate ) {
+		classes.push( baseClassName + '--indeterminate' );
+	}
+
+	if ( props.error ) {
+		classes.push( baseClassName + '--error' );
+	}
+
+	// Removing non-native attributes before passing it to the Checkbox component.
+	delete attrs.indeterminate;
```
Comment (rotemee) [inline core/app/assets/js/ui/atoms/checkbox.js:23]: If attrs can be removed and pass only the 'checked' prop.

File: core/app/assets/js/ui/atoms/checkbox.js [hunk 3]
```diff
@@ -23,10 +37,13 @@ export default function Checkbox( props ) {
...
 	disabled: PropTypes.bool,
+	indeterminate: PropTypes.bool,
 	rounded: PropTypes.bool,
+	error: PropTypes.bool,
 };
 
 Checkbox.defaultProps = {
 	className: '',
 	disabled: false,
+	indeterminate: false,
```
Comment (rotemee) [inline core/app/assets/js/ui/atoms/checkbox.js:48]: error: false

File: core/app/assets/js/ui/table/table-checkbox.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	// Removing non-native attributes before passing it to the Checkbox component.
+	delete attrs.allSelectedCount;
+
+	return (
+		<Checkbox
+			checked={ getIsSelected() }
+			indeterminate={ isIndeterminate() }
+			onChange={ onChange }
+			disabled={ getIsDisabled() }
+			{ ...attrs }
```
Comment (rotemee) [inline core/app/assets/js/ui/table/table-checkbox.js:55]: Make sure that it's needed.

File: core/app/assets/js/ui/table/table.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+export default function Table( props ) {
+	const getInitialSelections = () => {
+		const initialSelected = {};
+
+		props.initialSelected.forEach( ( value ) => initialSelected[ value ] = value );
+
+		return initialSelected;
+	},
+	[ selected, setSelected ] = useState( getInitialSelections() ),
```
Comment (rotemee) [inline core/app/assets/js/ui/table/table.js:22]: Check optimiation.

File: core/app/assets/js/utils/utils.js [hunk 6]
```diff
@@ -37,3 +37,11 @@ export const rgbToHex = ( r, g, b ) => '#' + [ r, g, b ].map( ( x ) => {
 export const isOneOf = ( filetype, filetypeOptions ) => {
 	return filetypeOptions.some( ( type ) => filetype.includes( type ) );
 };
+
+export const arrayToObjectByKey = ( array = [], key ) => {
```
Comment (rotemee) [inline core/app/assets/js/utils/utils.js:41]: Remove array default.

File: core/app/modules/import-export/assets/js/hooks/use-plugins.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+		pluginsActions: {
+			fetch: fetchData,
+			install,
+			activate,
+			deactivate,
+			remove,
+			reset,
+		},
+		PLUGINS_RESPONSE_MAP,
+		PLUGIN_STATUS_MAP,
```
Comment (rotemee) [inline core/app/modules/import-export/assets/js/hooks/use-plugins.js:126]: Remove.

File: core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import PluginsSelection from '../../../../../shared/plugins-selection/plugins-selection';
+import Loader from '../../../../../ui/loader/loader';
+
+import usePlugins, { PLUGIN_STATUS_MAP } from '../../../../../hooks/use-plugins';
+import usePluginsData, { PLUGINS_KEYS } from '../../../../../hooks/use-plugins-data';
+
+export default function ExportPluginsSelection() {
+	const exportContext = useContext( ExportContext ),
+		{ response } = usePlugins(),
+		{ pluginsData } = usePluginsData( response.data ),
```
Comment (rotemee) [inline core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js:14]: Check for re-render (if need to add useMemo).

File: core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+		activePlugins = pluginsData.filter( ( { status } ) => PLUGIN_STATUS_MAP.ACTIVE === status ),
+		handleOnSelect = ( selectedPlugins ) => exportContext.dispatch( { type: 'SET_PLUGINS', payload: selectedPlugins } ),
+		getInitialSelected = () => {
+			// Elementor Core will always be th
```
Comment (rotemee) [inline core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js:18]: fix comment (will always be the first plugin).

File: core/app/assets/js/molecules/data-table.js (hunk 1 above)
Comment (rotemee) [reply core/app/assets/js/molecules/data-table.js:14]: Done (removed).

File: core/app/assets/js/ui/atoms/checkbox.js (hunk 2 above)
Comment (rotemee) [reply core/app/assets/js/ui/atoms/checkbox.js:23]: Done (removed).

File: core/app/assets/js/ui/atoms/checkbox.js (hunk 3 above)
Comment (rotemee) [reply core/app/assets/js/ui/atoms/checkbox.js:48]: Done.

File: core/app/assets/js/ui/table/table-checkbox.js (hunk 4 above)
Comment (rotemee) [reply core/app/assets/js/ui/table/table-checkbox.js:55]: Done (removed).

File: core/app/assets/js/ui/table/table.js (hunk 5 above)
Comment (rotemee) [reply core/app/assets/js/ui/table/table.js:22]: Done (the 'selected' was changed from Object to an Array, therefore the function is no longer needed and was removed).

File: core/app/assets/js/utils/utils.js (hunk 6 above)
Comment (rotemee) [reply core/app/assets/js/utils/utils.js:41]: Done.

File: core/app/modules/import-export/assets/js/hooks/use-plugins.js (hunk 7 above)
Comment (rotemee) [reply core/app/modules/import-export/assets/js/hooks/use-plugins.js:126]: Done.

File: core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js (hunk 8 above)
Comment (rotemee) [reply core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js:14]: Changed the structure of the components + prevented the entire component re-render.

File: core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js (hunk 9 above)
Comment (rotemee) [reply core/app/modules/import-export/assets/js/pages/export/export-plugins/components/export-plugins-selection/export-plugins-selection.js:18]: Fixed.

File: core/app/modules/import-export/assets/js/pages/import/import-content/components/import-content-display/import-content-display.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import InlineLink from 'elementor-app/ui/molecules/inline-link';
+
+export default function ImportContentDisplay( {
+	manifest,
+	hasPro,
+	hasPlugins,
+	isAllRequiredPluginsSelected,
+	onResetProcess,
+} ) {
+	const contentData = kitContentData.filter( ( { type } ) => {
```
Comment (rotemee) [inline core/app/modules/import-export/assets/js/pages/import/import-content/components/import-content-display/import-content-display.js:13]: Change contentData name + add comment.

Comment (rotemee) [reply core/app/modules/import-export/assets/js/pages/import/import-content/components/import-content-display/import-content-display.js]: Done.

**Review by yossi-elementor — APPROVED** [review]

**Pull request closed without merging**